File: lib/Model/product.dart
import 'package:flutter/material.dart';

class Product {
  String description, title, imageName;
  int _id, storeId, price;

  Image? image;

  Product(this.description, this._id, this.title, this.imageName, this.storeId,
      this.price);
  int get id {
    return _id;
  }

  Product.fromJson(Map<String, dynamic> json)
      : _id = json['_id'],
        storeId = json['storeId'],
        price = json['price'],
        description = json['description'],
        title = json['title'],
        imageName = json['imageName'];
}

File: server/model/product.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const productSchema = new Schema({
    _id: { type: Number, required: true },
    title: { type: String, required: true },
    description: { type: String, required: true },
    price: { type: Number, required: true },
    storeId: { type: Number, required: true },
    imageName: { type: String, required: true },
}, { collection: 'product' });

module.exports = mongoose.model('Product', productSchema);
